packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx

```diff
@@ -63,100 +64,259 @@ type ExpandableBlockProps = {
   disabled?: boolean;
 };
 
-/**
- * Container that allows content to be hidden behind a brief title and a caption.
- * @example
- * <ExpandableBlock title='Basic Block'>Content in block!</ExpandableBlock>
- * <ExpandableBlock title='Basic Block' caption='basic caption'>Content in block!</ExpandableBlock>
- * <ExpandableBlock title='Default Expanded Block' caption='basic caption' isExpanded={true}>Content in block!</ExpandableBlock>
- * <ExpandableBlock title='Positive status' status='positive'>Content</ExpandableBlock>
- * <ExpandableBlock title='Block with icon' endIcon={<SvgPlaceholder />}>Content</ExpandableBlock>
- */
-export const ExpandableBlock = React.forwardRef((props, ref) => {
+const ExpandableBlockComponent = React.forwardRef((props, forwardedRef) => {
   const {
-    caption,
     children,
     className,
-    title,
     onToggle,
     style,
-    isExpanded = false,
-    endIcon,
+    isExpanded,
     status,
     size = 'default',
     styleType = 'default',
     disabled = false,
     ...rest
   } = props;
 
-  const icon = endIcon ?? (status && StatusIconMap[status]());
-
-  const [expanded, setExpanded] = React.useState(isExpanded);
-  React.useEffect(() => {
-    setExpanded(isExpanded);
-  }, [isExpanded]);
-
-  const handleToggle = () => {
-    if (disabled) {
-      return;
-    }
-    setExpanded(!expanded);
-    onToggle?.(!expanded);
-  };
-
-  const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
-    if (event.altKey || disabled) {
-      return;
-    }
-
-    if (
-      event.key === 'Enter' ||
-      event.key === ' ' ||
-      event.key === 'Spacebar'
-    ) {
-      handleToggle();
-    }
-  };
+  const [expandedState, setExpanded] = React.useState(isExpanded);
+  const expanded = isExpanded ?? expandedState;
+
+  return (
+    <ExpandableBlockContext.Provider
+      value={{
+        status,
+        isExpanded: expanded,
+        onToggle,
+        size,
+        styleType,
+        disabled,
+        setExpanded,
+        children,
+      }}
+    >
+      <Box
+        className={cx(
+          'iui-expandable-block',
+          {
+            'iui-expanded': expanded,
+            'iui-small': size === 'small',
+            'iui-borderless': styleType === 'borderless',
```

Comment:
`iui-expanded`, `iui-small` and `iui-borderless` could be data attributes

https://github.com/iTwin/iTwinUI/pull/1354/files/924a3064d78e60073950372b6138023367570f5d..001d046d306da6846b67f7765cf7573715ce348f#r1246683447

Reply:
these class names have been updated :d

Reply:
i was thinking converting them to data attributes would be better than class names, like so:

```html
<div class='iui-expandable-block' data-iui-size='small' data-iui-expanded='false' data-iui-disabled='true'>
```

you can see an example in `button.scss`+`Button.tsx`

Reply:
ohhh o_o

Reply:
Done! had some trouble with fixing tests (that also came from other changes not just this in particular), but @adamhock helped me :D